File: CPlusPlus/Command/src/Waiter.cpp
/*************************************************************************
    > File Name: Waiter.cpp
    > Author: ma6174
    > Mail: ma6174@163.com 
    > Created Time: 2017年04月09日 星期日 10时32分42秒
 ************************************************************************/

#include"Waiter.h"

using namespace std;

void Waiter::SetCommand(const string& name, const shared_ptr<Command>& command)
{
	mapping[name].push_back(command);
}

void Waiter::CancleCommand(const string& name, const shared_ptr<Command>& command)
{
	if(mapping.find(name)!=mapping.end())
	{
		mapping[name].remove(command);
	}
}

void Waiter::Notify(const string& name)
{
	for(auto it=mapping[name].begin();it!=mapping[name].end();++it)
	{
		(*it)->Execute();
	}
}

File: Python/Command/Waiter.py
# -*- coding:utf-8 -*-

from Command import MakeMuttonCommand
from Command import MakeChickenWingCommand

class Waiter:

    def __init__(self):
        self.__mapping = dict()

    def SetCommand(self, name, command):
        if name not in self.__mapping.keys():
            self.__mapping[name] = []
        self.__mapping[name].append(command)

    def CancleCommand(self, name, command):
        if name in self.__mapping.keys():
            self.__mapping[name].remove(command)

    def Notify(self, name):
        if name in self.__mapping:
            for cmd in self.__mapping[name]:
                cmd.Execute()

        
